### ISSUE TITLE
Compile Error: 'zend_arg_info' {aka ‘struct _zend_arg_info’} has no member named 'pass_by_reference'

### ISSUE THREAD

**Issue by ammarfaizi2** [krakjoe/parallel#106]
Hi,
I am sorry if my question looks like trivial since I do not have a lot of experiences with PHP core.

So, I tried to clone [php-src](https://github.com/php/php-src) and compile it. It went fine, but when I compiled the parallel, it seems to be having a problem here.

Does this mean that parallel is currently not supported in PHP 8?
Or maybe I am missing something here?

```
# Compile PHP
git clone https://github.com/php/php-src --depth 1
cd php-src
./buildconf
./configure --enable-sigchild --enable-libgcc --enable-dmalloc --enable-dtrace --enable-fd-setsize --enable-werror --with-valgrind --with-mysqli --with-mysql-sock --enable-intl --with-gmp --with-gettext --with-jpeg --with-webp --with-xpm --with-freetype --with-external-gd --enable-gd --with-ffi --with-pcre-jit --enable-calendar --enable-dba --with-openssl --with-kerberos --with-system-ciphers --with-enchant --with-zip --enable-mysqlnd --enable-zts --with-mhash --enable-fpm --enable-sysvsem --enable-sysvshm --enable-sysvmsg --enable-sockets --enable-pcntl --with-sodium --enable-soap --with-readline
make
sudo make install

# Compile parallel
cd ext
git clone https://github.com/krakjoe/parallel
cd parallel
phpize
./configure
make
```

```
libtool: compile:  cc -Wall -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -I. -I/home/ammarfaizi2/project/now/php-src/ext/parallel -DPHP_ATOM_INC -I/home/ammarfaizi2/project/now/php-src/ext/parallel/include -I/home/ammarfaizi2/project/now/php-src/ext/parallel/main -I/home/ammarfaizi2/project/now/php-src/ext/parallel -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -I/home/ammarfaizi2/project/now/php-src/ext/parallel -DHAVE_CONFIG_H -g -O2 -c /home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c  -fPIC -DPIC -o src/.libs/check.o
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c: In function ‘php_parallel_check_arginfo’:
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c:129:44: warning: implicit declaration of function ‘ZEND_TYPE_CODE’; did you mean ‘ZEND_TYPE_CE’? [-Wimplicit-function-declaration]
  129 |         if (ZEND_TYPE_IS_SET(it->type) && (ZEND_TYPE_CODE(it->type) == IS_OBJECT || ZEND_TYPE_IS_CLASS(it->type))) {
      |                                            ^~~~~~~~~~~~~~
      |                                            ZEND_TYPE_CE
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c:129:85: warning: implicit declaration of function ‘ZEND_TYPE_IS_CLASS’; did you mean ‘ZEND_TYPE_HAS_CLASS’? [-Wimplicit-function-declaration]
  129 |         if (ZEND_TYPE_IS_SET(it->type) && (ZEND_TYPE_CODE(it->type) == IS_OBJECT || ZEND_TYPE_IS_CLASS(it->type))) {
      |                                                                                     ^~~~~~~~~~~~~~~~~~
      |                                                                                     ZEND_TYPE_HAS_CLASS
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c:139:15: error: ‘zend_arg_info’ {aka ‘struct _zend_arg_info’} has no member named ‘pass_by_reference’
  139 |         if (it->pass_by_reference) {
      |               ^~
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c:165:15: error: ‘zend_arg_info’ {aka ‘struct _zend_arg_info’} has no member named ‘pass_by_reference’
  165 |         if (it->pass_by_reference) {
      |               ^~
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c: In function ‘php_parallel_check_class_inline’:
/home/ammarfaizi2/project/now/php-src/ext/parallel/src/check.c:538:13: warning: implicit declaration of function ‘ZEND_TYPE_IS_CE’; did you mean ‘ZEND_TYPE_IS_SET’? [-Wimplicit-function-declaration]
  538 |         if (ZEND_TYPE_IS_CE(info->type)) {
      |             ^~~~~~~~~~~~~~~
      |             ZEND_TYPE_IS_SET
make: *** [Makefile:209: src/check.lo] Error 1
```

**Comment by krakjoe** [comment]
Currently support for PHP 8 is not complete, since changes were merged for union types.

**Comment by ammarfaizi2** [comment]
Ref: https://github.com/krakjoe/parallel/issues/156

**Comment by krakjoe** [comment]
resolved by be6d83e8f9ef3fdc5ed773c0d1c574205d666ca1

**Comment by ttodua** [comment]
I still have same issue, using latest Debian buster. `pecl install parallel` ends up with same
```
/tmp/pear/temp/parallel/src/check.c: In function 'php_parallel_check_arginfo':
/tmp/pear/temp/parallel/src/check.c:129:44: warning: implicit declaration of function 'ZEND_TYPE_CODE'; did you mean 'ZEND_TYPE_CE'? [-Wimplicit-function-declaration]
         if (ZEND_TYPE_IS_SET(it->type) && (ZEND_TYPE_CODE(it->type) == IS_OBJECT || ZEND_TYPE_IS_CLASS(it->type))) {
                                            ^~~~~~~~~~~~~~
                                            ZEND_TYPE_CE
/tmp/pear/temp/parallel/src/check.c:129:85: warning: implicit declaration of function 'ZEND_TYPE_IS_CLASS'; did you mean 'ZEND_TYPE_HAS_CLASS'? [-Wimplicit-function-declaration]
         if (ZEND_TYPE_IS_SET(it->type) && (ZEND_TYPE_CODE(it->type) == IS_OBJECT || ZEND_TYPE_IS_CLASS(it->type))) {
                                                                                     ^~~~~~~~~~~~~~~~~~
                                                                                     ZEND_TYPE_HAS_CLASS
/tmp/pear/temp/parallel/src/check.c:139:15: error: 'zend_arg_info' {aka 'struct _zend_arg_info'} has no member named 'pass_by_reference'
         if (it->pass_by_reference) {
               ^~
/tmp/pear/temp/parallel/src/check.c:165:15: error: 'zend_arg_info' {aka 'struct _zend_arg_info'} has no member named 'pass_by_reference'
         if (it->pass_by_reference) {
               ^~
/tmp/pear/temp/parallel/src/check.c: In function 'php_parallel_check_class_inline':
/tmp/pear/temp/parallel/src/check.c:538:13: warning: implicit declaration of function 'ZEND_TYPE_IS_CE'; did you mean 'ZEND_TYPE_IS_SET'? [-Wimplicit-function-declaration]
         if (ZEND_TYPE_IS_CE(info->type)) {
             ^~~~~~~~~~~~~~~
             ZEND_TYPE_IS_SET
make: *** [Makefile:224: src/check.lo] Error 1
ERROR: `make' failed
```

Does anyone have same problem or how you solved.
-- 
Update: I've ended up installing with `git & make` as suggested in install notes.
